### PR TITLE
Proposed codestyle

### PR DESCRIPTION
**Pull request by Kostronor**
Changed braces style and added some improvements.

### CODE REVIEW

File: src/org/company/MyClass.java [hunk 1]
```diff
@@ -5,11 +5,9 @@
 /**
  * The {@link MyClass} is a simple class.
  *
- * @author MyName
  * @since Current.Project.Version
  */
-public class MyClass extends AbstractMyClass
-{
+public class MyClass extends AbstractMyClass{
```
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: space before the brace.

File: src/org/company/MyClass.java [hunk 2]
```diff
@@ -35,31 +32,43 @@ public MyClass( )
...
         // Nothing
     }
 
     /**
      * This methods froms from {@link org.company.interfaces.IMyInterface}
      */
+    @NotNull
+    public String makeItWork(@Nullable String foo, @NotNull String bar){
+        // Check for null on @Nullable
+        if(foo == null){
```
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: space before the brace

File: src/org/company/MyClass.java [hunk 3]
```diff
@@ -35,31 +32,43 @@ public MyClass( )
...
+            //Early return on null checks and important requirements
+            return "";
+        }
+        
+        //@NotNull requires the caller to check for null
+        if(bar.length > 0){
```
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: !bar.isEmpty() :shipit:

File: src/org/company/MyClass.java [hunk 4]
```diff
@@ -35,31 +32,43 @@ public MyClass( )
...
-    {
-        if ( checkCondition( ) )
-        {
+    public void doSomething(){
+        if(checkCondition()){
             // Do something
-        }
-        else
-        {
+        }else{
```
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: } else {

File: src/org/company/MyClass.java (hunk 3 above)
Comment (Isfirs) [inline src/org/company/MyClass.java]: I prefer using methods over manual conditions, too.

File: src/org/company/MyClass.java (hunk 4 above)
Comment (Isfirs) [inline src/org/company/MyClass.java]: I think keeping `else if` & `else` on different lines then the braces.

File: src/org/company/abstracts/AbstractMyClass.java [hunk 5]
```diff
@@ -2,15 +2,15 @@
 
 import org.company.interfaces.IMyInterface;
 
-public abstract class AbstractMyClass implements IMyInterface{
+public abstract class AbstractMyClass implements IMyInterface {
     
     private int localInt;
```
Comment (Isfirs) [inline src/org/company/abstracts/AbstractMyClass.java]: Do table alignment.

Comment (Kostronor) [inline src/org/company/abstracts/AbstractMyClass.java]: Not manually over github web :tongue:

File: src/org/company/MyClass.java [hunk 6]
```diff
@@ -25,8 +23,7 @@
      *
      * Doing nothing.
      */
-    public MyClass( )
-    {
+    public MyClass() {
```
Comment (Isfirs) [inline src/org/company/MyClass.java]: No whitespace in empty argument body? It feels ugly for me.

File: src/org/company/MyClass.java [hunk 7]
```diff
@@ -35,31 +32,43 @@ public MyClass( )
...
      * A method doing exactly nothing.
      */
-    public void doNothing( )
-    {
+    public void doNothing() {
         // Nothing
     }
 
     /**
      * This methods froms from {@link org.company.interfaces.IMyInterface}
```
Comment (Isfirs) [inline src/org/company/MyClass.java]: If we leave this empty, the jdoc compiler will automatically take the parent one. I think, this is needless.

File: src/org/company/MyClass.java (hunk 6 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: The whitespace feels so unused... What does it help? what do we gain with it?

File: src/org/company/MyClass.java [hunk 8]
```diff
@@ -69,17 +78,15 @@ public void doSomething( )
...
-    {
-        return ( (this.anInt > 0) ? true : false);
+    private boolean checkCondition() {
+        return this.anInt > 0;
     }
 
-    public static void main(String...args)
-    {
-        MyClass myClass = new MyClass( );
+    public static void main(String... args) {
```
Comment (Isfirs) [inline src/org/company/MyClass.java]: The whitespace here feels "ugly" too. Most `list` expression I saw in the past did **not** consider a whitespace beween the periods and the variable as an identifier `this is a list argument`.

File: src/org/company/MyClass.java [hunk 9]
```diff
@@ -69,17 +78,15 @@ public void doSomething( )
...
+        MyClass myClass = new MyClass();
 
-        myClass.equals(new Object( ));
+        myClass.equals(new Object());
     }
 
 }
-// Empty line at the end!
\ No newline at end of file
+// Empty line at the end for git!
```
Comment (Isfirs) [inline src/org/company/MyClass.java]: For git? I think this is a design pattern to have a blank line at the end of the file.

File: src/org/company/MyClass.java (hunk 7 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: And if we don't have a parent one and this needs to be something empty? There is no @Override so there is no parent. And maybe others want to override this sometimes but not all the time. We want such methods to have a comment why they are empty

File: src/org/company/MyClass.java (hunk 8 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: Without the whitespace it feels weird, we should separate type `String...` and variable `args` clearly

File: src/org/company/MyClass.java (hunk 9 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: Nope, git needs it

File: src/org/company/MyClass.java [hunk 10]
```diff
@@ -5,11 +5,9 @@
 /**
  * The {@link MyClass} is a simple class.
  *
- * @author MyName
```
Comment (Isfirs) [inline src/org/company/MyClass.java]: How do we define/know, how was responsible at this class, if questions come up or something is wrong?

File: src/org/company/MyClass.java (hunk 9 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: http://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: src/org/company/MyClass.java (hunk 10 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: Git blame and we are all responsible. You can always ask on slack (and you know who did what etc.) We discussed this quite lengthy

File: src/org/company/MyClass.java (hunk 9 above)
Comment (Isfirs) [inline src/org/company/MyClass.java]: Ok, good to know. :)
But still, for me it is design pattern! 👍

Comment (Kostronor) [inline src/org/company/MyClass.java]: Maybe, but git really needs this so there are no unneeded merge conflicts and the like :D

File: src/org/company/MyClass.java (hunk 7 above)
Comment (Isfirs) [inline src/org/company/MyClass.java]: The jdoc compiler process searches for a local, `child` `javadoc` definition.

- If this is present, it compiles it as is. If you want to have the parent `javadoc` in it, you would have to use `@inheritDoc`. 
- If the compiler does not find a `javadoc` definition, it searches for a parent one.

I haven't seen the needs for `@Override` in the past.

File: src/org/company/MyClass.java (hunk 6 above)
Comment (Isfirs) [inline src/org/company/MyClass.java]: It's just some spacing to help the eye.

`() vs `( )` feels friendlier for me :D

File: src/org/company/MyClass.java (hunk 7 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: @Override is there so you don't accidentally not override http://stackoverflow.com/questions/8545540/why-override-needed-in-java-or-android
and you !have! to use it, sonar will shout at you if you don't :P

If we have that cleared up this method has no parent.

File: src/org/company/MyClass.java (hunk 6 above)
Comment (Kostronor) [inline src/org/company/MyClass.java]: `()` feels like "This is empty"
`( )` feels like "This has a reason to not be completely empty and I have to stop what I'm doing for a second to remember that java compilers are smart and we are not in js/python where this changes syntax"

Comment (cltnschlosser) [inline src/org/company/MyClass.java]: `( )` I feel like this leaves the closing parenthesis floating out there, really not a fan of that.

File: src/org/company/MyClass.java (hunk 10 above)
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: If the code review process is working as it should, then yes as a team we are responsible.

File: src/org/company/MyClass.java [hunk 11]
```diff
@@ -35,31 +32,43 @@ public MyClass( )
...
+    /**
+     * This methods froms from {@link org.company.interfaces.IMyInterface}
+     */
     @Override
-    public void doSomething( )
-    {
-        if ( checkCondition( ) )
-        {
+    public void doSomething() {
+        if(checkCondition()) {
```
Comment (cltnschlosser) [inline src/org/company/MyClass.java]: I would do `if (checkCondition()) {`

**Pull request merged**